src/main/java/com/capitalone/dashboard/collector/DefaultSonar6Client.java

```diff
@@ -64,34 +66,44 @@
 
     @Autowired
     public DefaultSonar6Client(RestClient restClient, SonarSettings settings) {
-        if(StringUtils.isEmpty(settings.getUsername())|| StringUtils.isEmpty(settings.getPassword())){
-            this.userInfo= new RestUserInfo("","");
-        }else{
-            this.userInfo = new RestUserInfo(settings.getUsername(),settings.getPassword());
-        }
-
         this.restClient = restClient;
 
         // override default sonar metrics to fetch via properties file settings
         if (!StringUtils.isEmpty(settings.getMetrics63andAbove())) {
             metrics = settings.getMetrics63andAbove();
+        } else {
+            metrics = DEFAULT_METRICS;
+        }
+    }
+
+    @Override
+    public void setServerCredentials(String username, String password, String token) {
+        // use token when given
```

Comment:
we need to handle the if both the information is given then provide an error stating only 1 mode of authentication is needed. either token or username/password